testutil/integration.go

```diff
@@ -162,14 +169,80 @@ func DeliverTx(
 		return abci.ResponseDeliverTx{}, err
 	}
 
+	return broadcastTxBytes(appEvmos, encodingConfig.TxConfig.TxEncoder(), txBuilder.GetTx())
+}
+
+// DeliverEthTx generates and broadcasts a Cosmos Tx populated with MsgEthereumTx messages.
+// If a private key is provided, it will attempt to sign all messages with the given private key,
+// otherwise, it will assume the messages have already been signed.
+func DeliverEthTx(
+	ctx sdk.Context,
+	appEvmos *app.Evmos,
+	priv *ethsecp256k1.PrivKey,
+	msgs ...sdk.Msg,
+) (abci.ResponseDeliverTx, error) {
+	encodingConfig := encoding.MakeConfig(app.ModuleBasics)
+	denom := appEvmos.ClaimsKeeper.GetParams(ctx).ClaimsDenom
+
+	txBuilder := encodingConfig.TxConfig.NewTxBuilder()
+
+	signer := ethtypes.LatestSignerForChainID(appEvmos.EvmKeeper.ChainID())
+	txFee := sdk.Coins{}
+	txGasLimit := uint64(0)
+
+	// Sign messages and compute gas/fees.
+	for _, m := range msgs {
+		msg, ok := m.(*evmtypes.MsgEthereumTx)
+		if !ok {
+			return abci.ResponseDeliverTx{}, errorsmod.Wrapf(errorsmod.Error{}, "cannot mix Ethereum and Cosmos messages in one Tx")
+		}
+
+		if priv != nil {
+			err := msg.Sign(signer, tests.NewSigner(priv))
+			if err != nil {
+				return abci.ResponseDeliverTx{}, err
+			}
+		}
+
+		msg.From = ""
```

Comment:
Q: Why the `From` field is updated to `From = ""`?

Reply:
The `From` field is expected to be empty on broadcast, see [EthValidateBasic](https://github.com/evmos/ethermint/blob/292fe03b2d1811360f3297ed81bb088a432a04e8/app/ante/setup.go#L171)